pkg/reconciler/tenancy/clusterworkspacedeletion/deletion/workspace_resource_deletor.go

```diff
@@ -463,16 +454,18 @@ func (d *workspacedResourcesDeleter) deleteAllContent(ctx context.Context, ws *t
 		}
 		// sort for stable updates
 		sort.Strings(remainingByFinalizer)
+		contentDeletedMessages = append(contentDeletedMessages, fmt.Sprintf("Some content in the workspace has finalizers remaining: %s", strings.Join(remainingByFinalizer, ", ")))
+	}
+
+	if len(contentDeletedMessages) > 0 {
 		conditions.MarkFalse(
 			ws,
 			tenancyv1alpha1.WorkspaceContentDeleted,
-			"SomeFinalizersRemain",
+			"SomeResourcesRemain",
 			conditionsv1alpha1.ConditionSeverityError,
-			fmt.Sprintf("Some content in the workspace has finalizers remaining: %s", strings.Join(remainingByFinalizer, ", ")),
+			strings.Join(contentDeletedMessages, "; "),
```

Comment:
is this to ensure the condition message stable?

Reply:
per iteration one must never call `conditions.MarkFalse/True` twice because it will hotloop. Hence, I join both messages here and do one call only.

Reply:
no, it's by design. It just compares reason+message and update the timestamp if one changes. But of course, if you call the func multiple times with different args (even if it converges to the same values every time) it will think there are updates.